opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -24,35 +24,50 @@ option java_multiple_files = true;
 option java_package = "io.opentelemetry.proto.trace.v1";
 option java_outer_classname = "TraceProto";
 
-// A span represents a single operation within a trace. Spans can be
+// Span represents a single operation within a trace. Spans can be
 // nested to form a trace tree. Spans may also be linked to other spans
-// from the same or different trace. And form graphs. Often, a trace
+// from the same or different trace and form graphs. Often, a trace
 // contains a root span that describes the end-to-end latency, and one
 // or more subspans for its sub-operations. A trace can also contain
 // multiple root spans, or none at all. Spans do not need to be
 // contiguous - there may be gaps or overlaps between spans in a trace.
 //
-// The next id is 17.
+// The next field id is 18.
 message Span {
-  // A unique identifier for a trace. All spans from the same trace share
-  // the same `trace_id`. The ID is a 16-byte array. An ID with all zeroes
-  // is considered invalid.
+  // trace_id is the unique identifier of a trace. All spans from the same trace share
+  // the same `trace_id`.
   //
-  // This field is semantically required. Receiver should generate new
-  // random trace_id if empty or invalid trace_id was received.
+  // This field is semantically required. If an invalid trace ID was received:
+  // - The receiver MAY reject the invalid data and respond with the appropriate error
+  //   code to the sender.
+  // - The receiver MAY accept the invalid data and attempt to correct it.
   //
-  // This field is required.
-  bytes trace_id = 1;
-
-  // A unique identifier for a span within a trace, assigned when the span
-  // is created. The ID is an 8-byte array. An ID with all zeroes is considered
-  // invalid.
+  // The ID is a 16-byte value represented using two fixed64 fields. An ID with
+  // value of 0 for all bytes is considered invalid.
+  // To convert this pair of fixed64 fields to an ID represented as a 16-byte array put
```

Comment:
is little-endian used in other systems like Jaeger and Zipkin?

Reply:
Jaeger uses big endian, evidently: https://github.com/jaegertracing/jaeger/blob/master/model/ids.go#L81

Reply:
I would prefer to be the consistent with Jaeger :)

Reply:
according to wikipedia:

> big-endianness is the dominant ordering in networking protocols (IP, TCP, UDP)

Reply:
And performance wise even though Intel is little endian the transformation is an operation that is super fast (no transistors involved) change a wire reverse.

Reply:
ProtoBuf uses little-endian encoding. Also most (?) server CPUs today use little endian in-memory representation. This means on little-endian CPUs ProtoBuf does straight copying of fixed64 fields when serializing. Requiring big-endian for this field will require extra endianness conversion on little-endian CPUs.

Reply:
So while I agree it would be nice to be consistent with Jaeger it will also mean a performance hit. I can benchmark and see if it is a significant performance hit.

Reply:
There is no measurable performance difference so I am changing to use big endian byte order so that we are consistent with Jaeger.

Reply:
@SergeyKanzhelev I changed to big endian, explanation above. Please unblock and merge.